### PR TITLE
Allow multiple python plugin variable objects in idd

### LINKED ISSUE #9227: PythonPlugin:Variables multiple objects issue
Issue overview
--------------
The EnergyPlus takes only one PythonPlugin:Variables object and gives error if there are more than one object. This means maximum of 10 variables can be defined (as shown in image attached) and used for getting output.
The EnergyPlus file with only one  PythonPlugin:Variables works completely fine.
[Eplus_error.zip](https://github.com/NREL/EnergyPlus/files/7822610/Eplus_error.zip)

### Details
Some additional details for this issue (if relevant):
 - Platform (Windows)
 - Version of EnergyPlus (9.6)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 [ ] EnergyPlus error file added

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
 - Fixes #9227 

Note that no code changes were required.  The code was already set up to process multiple objects, the IDD was just limited with the unique object attribute.  I added a unit test prior to making the change and it gets:

```
Testing started at 12:00 PM ...
/tst/EnergyPlus/unit/Fixtures/EnergyPlusFixture.cc:197: Failure
Expected equality of these values:
  expected_string
    Which is: ""
  stream_str
    Which is: "   ** Severe  ** <root>[PythonPlugin:Variables] - Object should have no more than 1 properties.\n"
```

...as expected.  The test passes with the unique object attribute removed from the IDD object.  This should be an easy review, but I can add more stuff here if needed.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -104714,7 +104714,6 @@ PythonPlugin:Instance,
 PythonPlugin:Variables,
         \memo This object defines name identifiers for custom Python Plugin variable data that should be shared
         \memo among all running Python Plugins.
-        \unique-object
```
Comment (Myoldmopar) [inline idd/Energy+.idd.in]: No longer unique.

File: tst/EnergyPlus/unit/PluginManager.unit.cc [hunk 2]
```diff
@@ -93,4 +93,13 @@ TEST_F(EnergyPlusFixture, TestTrendVariable)
...
 }
+
+
+TEST_F(EnergyPlusFixture, MultiplePluginVariableObjects)
+{
+    std::string const idf_objects = ("PythonPlugin:Variables, Variables1, VariableA;  PythonPlugin:Variables, Variables2, VariableB;");
+    ASSERT_TRUE(process_idf(idf_objects));
+    PluginManagement::PluginManager p{*this->state};
+
+}
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/PluginManager.unit.cc:105]: New unit test failed when added to develop, passes with this branch.

**Comment by bonnema** [comment]
What happens if the same variable is repeated in two different objects? Could possibly happen by accident.

**Comment by Myoldmopar** [comment]
> What happens if the same variable is repeated in two different objects? Could possibly happen by accident.

Great point, I can add code in there to just ignore duplicated variables.  Or maybe a gentle warning would be better than just ignoring it.

**Comment by bonnema** [comment]
I’d vote for a warning. Maybe something like when you have a duplicate handle?

**Comment by Myoldmopar** [comment]
It's done: 

```
   ** Warning ** Found duplicate variable name in PythonPlugin:Variables objects, ignoring: "VariableA"
```

That work?

**Comment by bonnema** [comment]
Sweet!

**Comment by Myoldmopar** [comment]
All good here, merging this.

**Pull request merged**